evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -54,3 +58,67 @@ def transform(self, X, y=None):
         if y is not None:
             y = infer_feature_types(y)
         return X, y
+
+
+class BaseOverSampler(BaseSampler):
+    """Base Oversampler component. Used as the base class of all imbalance-learn oversampler components"""
+
+    def __init__(self, sampler, parameters, component_obj, random_seed):
+        error_msg = "imbalanced-learn is not installed. Please install using 'pip install imbalanced-learn'"
+        im = import_or_raise("imblearn.over_sampling", error_msg=error_msg)
+        self.sampler = {"SMOTE": im.SMOTE,
+                        "SMOTENC": im.SMOTENC,
+                        "SMOTEN": im.SMOTEN}[sampler]
+        super().__init__(parameters=parameters,
+                         component_obj=None,
+                         random_seed=random_seed)
+
+    def fit(self, X, y):
+        """Fits the Oversampler to the data.
+
+        Arguments:
+            X (ww.DataFrame): Training features
+            y (ww.DataColumn): Target features
+
+        Returns:
+            self
+        """
+        super().fit(X, y)
+        self._initialize_oversampler(X, y, self.sampler)
+
+    def _initialize_oversampler(self, X, y, sampler_class):
+        """Initializes the oversampler with the given sampler_ratio or sampler_ratio_dict. If a sampler_ratio_dict is provided, we will opt to use that.
+        Otherwise, we use will create the sampler_ratio_dict dictionary.
+
+        Arguments:
+            X (ww.DataFrame): Training features
+            y (ww.DataColumn): Target features
+            sampler_class (imblearn.BaseSampler): The sampler we want to initialize
+        """
+        _, _, _, y_pd = self._prepare_data(X, y)
+        sampler_params = {k: v for k, v in copy.copy(self.parameters).items() if k not in ['sampling_ratio', 'sampling_ratio_dict']}
+        if self.parameters['sampling_ratio_dict'] is not None and len(self.parameters['sampling_ratio_dict']):
+            # dictionary provided and takes priority
+            sampler_params['sampling_strategy'] = self.parameters['sampling_ratio_dict']
+        else:
+            # no dictionary provided. We pass the float if we have a binary situation
```

Comment:
Can you explain this comment? Looks like `make_balancing_dictionary` is making a dictionary regardless of whether it's a binary or multiclass problem.

Reply:
Yep, forgot to change the comment after editing.